docs: add note and example for NBSP (#2524)
keymanapp/help.keyman.com: developer/language/guide/control-keys.md

Using control characters in the context in a [`readonly`
group](groups#readonly-clause) is safe; no special handling is required.

# Non-breaking space (NBSP) in context

> Non-breaking space (NBSP) is used in the digital world to prevent automatic line breaks from being inserted between two words. Said another way, it prevents two words being separated on to two separate lines. In some applications NBSP are returned in the context to Keyman. Therefore in this case NBSP should be considered in a rule that detecs a blank space.

Reviewer: ```suggestion
Non-breaking space (NBSP) is used in the digital world to prevent automatic line breaks
from being inserted between two words. Said another way, it prevents two words being 
separated on to two separate lines. In some applications NBSP are returned in the context 
to Keyman. Therefore in this case NBSP should be considered in a rule that detects a 
blank space.
```

(Wrap text, but also correcting a typo)

Reviewer 2: Probably mention the Unicode codepoint here as well (extending Marc's suggestion):

```suggestion
Non-breaking space (NBSP, U+00A0) is used in the digital world to prevent automatic line
breaks from being inserted between two words. Said another way, it prevents two words
being separated on to two separate lines. In some applications NBSP are returned in the
context to Keyman. Therefore in this case NBSP should be considered in a rule that detects
a blank space.
```

> A example taken from the greek_tonzio keyboard. This keyboard uses stores; see  [`store()`](../reference/store)

Reviewer: ```suggestion
A example taken from the [greek_tonzio keyboard](https://github.com/keymanapp/keyboards/tree/master/release/g/greek_tonizo). This keyboard uses stores; see  [`store()`](../reference/store):
```

Could link to https://github.com/keymanapp/keyboards/tree/master/release/g/greek_tonizo

```
c Characters usually found before words  These characters are Space, parenthesis, quotes etc and to this we add the nbsp 
store(beforeWordChars)  U+0020 U+0028 U+002D U+0027 \
                        U+0022 U+003A U+007B U+00AB \
                        U+00B6 U+00A9 U+000A U+00A0 

c These are the characters [ ἀ ἐ ἠ ἰ ὀ ὑ ὠ Ἀ Ἐ Ἠ Ἰ Ὀ Ὑ Ὠ]                        
store(fwnWithBreathing) U+1F00 U+1F10 U+1F20 U+1F30 U+1F40 U+1F51 U+1F60 \
                        U+1F08 U+1F18 U+1F28 U+1F38 U+1F48 U+1F59 U+1F68
```

> Then we can have a rule that adds the breathing when the encountering a "before word" character

Reviewer: ```suggestion
Then we can have a rule that adds the breathing when the encountering a "before word" character:
```

```
c ------- GENERAL RULE: AUTOMATIC PSILI AT BEGINNING OF WORD STARTING WITH VOWEL ------
any(beforeWordChars) + any(vowels) > context index(fwnWithBreathing,2)
```

> A simple example: when "E" is pressed output a ἐ after a space, and ouput ε otherwise.

Author: There are two examples here, I don't intend to keep both. One is a taken from a "real world" keyboard the other is simpler so as to focus on the solution. What do the reviewers think.

Reviewer: I would definitely keep both. But start with the simple example and then expand to a real-life example. The simple example helps authors understand how it works. Then the real-life example shows how to apply it.

Reviewer: ```suggestion
A simple example: when <kbd>e</kbd> is pressed, output an `ἐ` after a space, and output `ε` otherwise:
```

```
store(spaceChrs)  U+0020 U+00A0 
c This nbsp Unicode Character is U+00A0 and space is U+0020. 
+ [K_E] > U+03B5

> any(beforeWordChars) + [K_E] > context U+1f10

Reviewer 2: I guess this should be:

```suggestion
any(spaceChrs) + [K_E] > context U+1f10
```

Using control characters in the context in a [`readonly`
group](groups#readonly-clause) is safe; no special handling is required.

# Non-breaking space (NBSP) in context

Non-breaking space (NBSP, U+00A0) is used in the digital world to prevent automatic line
breaks from being inserted between two words. Said another way, it prevents two words
being separated on to two separate lines. In some applications NBSP are returned in the

> context to Keyman see issue with Firefox [#14945](https://github.com/keymanapp/keyman/issues/14945).

Reviewer: ```suggestion
context to Keyman (for example, see issue with Firefox [#14945](https://github.com/keymanapp/keyman/issues/14945)).
```

Author: This was on auto merge I will push a new PR.
